api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -84,4 +84,16 @@ RewriteFiles rewriteFiles(Set<DataFile> dataFilesToReplace, Set<DeleteFile> dele
    * @return this for method chaining
    */
   RewriteFiles validateFromSnapshot(long snapshotId);
+
+  /**
+   * Set the snapshot ID used in any reads for this operation.
+   * <p>
+   * Validations will check changes after this snapshot ID. If this is not called, all ancestor snapshots through the
+   * table's initial snapshot are validated.
+   *
+   * @param snapshotId a snapshot ID
+   * @param ignorePosDeletes whether to ignore position delete in validation
+   * @return this for method chaining
+   */
+  RewriteFiles validateFromSnapshot(long snapshotId, boolean ignorePosDeletes);
```

Comment:
I don't think this is the right way to configure this option. Why not add a new method since this is orthogonal to the snapshot you're validating from? Or is there a more appropriate `validate` method that could be updated instead?

Reply:
Make sense to me. Let me refactor this.